src/main/frontend/src/app/pages/release-graph/release-node.service.ts

```diff
@@ -889,36 +949,127 @@ export class ReleaseNodeService {
     return finalNodes;
   }
 
-  private applyTimelineSpacing(
-    nodes: ReleaseNode[],
-    clusterStartDate: Date,
-    minSpacing: number,
-    minSpacingInside: number,
-  ): ReleaseNode[] {
-    const spacedFinalNodes: ReleaseNode[] = [];
+  private findLatestStable(nodes: ReleaseNode[]): ReleaseNode | null {
+    const nonNightly = nodes.filter((n) => !this.isNightlyRelease(n.label));
+    if (nonNightly.length === 0) return null;
 
-    for (const node of nodes) {
-      const previousNode = spacedFinalNodes.at(-1);
+    return nonNightly.reduce((latest, node) => {
+      const latestVersion = this.getVersionInfo(latest);
+      const nodeVersion = this.getVersionInfo(node);
 
-      // Skip spacing adjustment for mini nodes - they need to maintain their offset from branch nodes
-      if (previousNode && !node.isMiniNode) {
-        const gap = node.position.x - previousNode.position.x;
+      if (!latestVersion || !nodeVersion) return latest;
 
-        const nodeInZone = node.publishedAt >= clusterStartDate;
-        const previousNodeInZone = previousNode.publishedAt >= clusterStartDate;
+      if (nodeVersion.major > latestVersion.major) return node;
+      if (nodeVersion.major < latestVersion.major) return latest;
 
-        let currentMinSpacing = minSpacing;
-        if (nodeInZone || previousNodeInZone || node.isCluster || previousNode.isCluster) {
-          currentMinSpacing = minSpacingInside;
-        }
+      if (nodeVersion.minor > latestVersion.minor) return node;
+      if (nodeVersion.minor < latestVersion.minor) return latest;
 
-        if (gap < currentMinSpacing) {
-          node.position.x = previousNode.position.x + currentMinSpacing;
-        }
-      }
+      if (nodeVersion.patch > latestVersion.patch) return node;
+      return latest;
+    });
+  }
+
+  private findLatestNightly(nodes: ReleaseNode[]): ReleaseNode | null {
+    const nightlies = nodes.filter((n) => this.isNightlyRelease(n.label));
+    if (nightlies.length === 0) return null;
+
+    return nightlies.reduce((latest, node) => {
+      return node.publishedAt > latest.publishedAt ? node : latest;
+    });
+  }
+
+  private findLastNonNightlyRelease(nodes: ReleaseNode[]): ReleaseNode | null {
+    const nonNightly = nodes.filter((n) => !this.isNightlyRelease(n.label));
+    if (nonNightly.length === 0) return null;
+
+    return nonNightly.reduce((latest, node) => {
+      return node.publishedAt > latest.publishedAt ? node : latest;
+    });
+  }
+
+  private isBranchSupported(nodes: ReleaseNode[]): boolean {
+    const firstRelease = nodes.find((n) => !this.isNightlyRelease(n.label));
+    if (!firstRelease) return false;
+
+    const versionInfo = this.getVersionInfo(firstRelease);
+    if (!versionInfo) return false;
+
+    const supportDates = this.getSupportEndDates(firstRelease);
+    if (!supportDates) return false;
+
+    return new Date() <= supportDates.securitySupportEnd;
```

Comment:
This is really interesting, normally you cant really do a comparison like this for Date objects it does.
I wonder if you know why (this is not a critique, just curiosity)